muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/rest/WorkspaceForumRESTService.java

```diff
@@ -399,16 +400,18 @@ public Response updateThread(@PathParam ("WORKSPACEENTITYID") Long workspaceEnti
     }
     
     if (sessionController.hasPermission(MuikkuPermissions.OWNER, forumThread) || sessionController.hasWorkspacePermission(ForumResourcePermissionCollection.FORUM_EDIT_WORKSPACE_MESSAGES, workspaceEntity)) {
-      if (!forumThread.getSticky().equals(updThread.getSticky()) || !forumThread.getLocked().equals(updThread.getLocked())) {
+      if (!forumThread.getSticky().equals(updThread.getSticky()) || forumThread.getLock() != LockForumThread.valueOf(updThread.getLock())) {
```

Comment:
Is it guaranteed that `updThread.getLock()` is not `null`? `valueOf` won't deal with `null` values.